torch/distributed/elastic/agent/server/api.py

```diff
@@ -424,8 +427,7 @@ class ElasticAgent(abc.ABC):
     @abc.abstractmethod
     def run(self, role: str = DEFAULT_ROLE) -> RunResult:
         """
-        Runs the agent, retrying the worker group on failures up to
-        ``max_restarts``.
+        Run the agent, retrying the worker group on failures up to ``max_restarts``.
```

Comment:
We could change here following the approach you are taking:
~~~
“””Runs the agent.

Supports retrying the worker group on failures up to ``max_restarts``.

…
~~~